spec.go

```diff
@@ -306,6 +306,34 @@ type PackageDependencies struct {
 	// running a command in the built container.
 	// See [TestSpec] for more information.
 	Test []string `yaml:"test,omitempty" json:"test,omitempty"`
+
+	// ExtraRepos is used to inject extra package repositories that may be used to
+	// satisfy package dependencies in various stages.
+	ExtraRepos PackageRepositories `yaml:"extra_repos,omitempty" json:"extra_repos,omitempty"`
+}
+
+// PackageRepositories is used to configure repositories to add to the respective
+// environment.
+type PackageRepositories struct {
+	// Packging lists all the repos to add before running the package build.
+	// Add repositories here if the repository needs to be used to satisfy a build
+	// dependency.
+	Packaging []PackageRepositoryConfig `yaml:"packaging,omitempty" json:"packaging,omitempty"`
+	// Install lists all the repos to add before installing the produced package.
+	// Add repositories here if the repository needs to be used to satisfy a
+	// runtime dependency.
+	// Note: Repositories listed here are only used to satisfy dependencies during
+	// install and the repo configs may not be included in the result.
+	Install []PackageRepositoryConfig `yaml:"install,omitempty" json:"install,omitempty"`
+}
+
+type PackageRepositoryConfig struct {
+	// Keys are the list of keys that need to be imported to use the configured
+	// repositories
+	Keys []Source `yaml:"keys,omitempty" json:"keys,omitempty"`
+	// Config list of repo configs to to add to the environment.  The format of
+	// these configs is going to be distro specific (e.g. apt/yum configs).
+	Config []Source `yaml:"config" json:"config"`
```

Comment:
Is it likely for one repository "unit" to need multiple config files? Just trying to understand why we need an array here

Reply:
Mainly that I don't want to box in what can be done.
It also allows multiple repos to be installed at the same time using the same keys.

Reply:
Got it, that makes sense!